sw/device/lib/dif/dif_aes_unittest.cc

```diff
@@ -374,28 +376,94 @@ TEST_F(DataTest, DataOut) {
                                            {AES_STATUS_OUTPUT_VALID_BIT, true},
                                        });
 
-  for (uint32_t i = 0; i < ARRAYSIZE(data_.data); ++i) {
-    ptrdiff_t offset = AES_DATA_OUT_0_REG_OFFSET + (i * sizeof(uint32_t));
-    EXPECT_READ32(offset, data_.data[i]);
-  }
+  setExpectedReadMultreg(AES_DATA_OUT_0_REG_OFFSET, data_.data,
+                         ARRAYSIZE(data_.data));
 
   dif_aes_data_t out;
   EXPECT_DIF_OK(dif_aes_read_output(&aes_, &out));
 
   EXPECT_THAT(out.data, ElementsAreArray(data_.data));
 }
 
+class DataProcessTest : public AesTestInitialized {
+ protected:
+  const dif_aes_data_t dataIn[3] = {
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+  };
+
+  const dif_aes_data_t dataOut[3] = {
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+  };
+};
+
+TEST_F(DataProcessTest, ThreeBlocksSuccess) {
+  {
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_INPUT_READY_BIT, true},
+                                          {AES_STATUS_OUTPUT_VALID_BIT, true}});
+
+    setExpectedWriteMultreg(AES_DATA_IN_0_REG_OFFSET, dataIn[0].data,
+                            ARRAYSIZE(dataIn[0].data));
```

Comment:
Instead of using `ARRAYSIZE`, maybe it makes sense to have a third constant which is used in declaring the block count of the input and output, i.e. `static constexpr dif_aes_data_t kDataIn[kBlockCount];`